server/datastore/mysql/activities.go

```diff
@@ -155,6 +155,37 @@ func (ds *Datastore) ListActivities(ctx context.Context, opt fleet.ListActivitie
 	}
 	opt.ListOptions.IncludeMetadata = !(opt.ListOptions.UsesCursorPagination())
 
+	if opt.ListOptions.MatchQuery != "" {
+		activitiesQ += " AND (a.user_name LIKE ? OR a.user_email LIKE ?)"
+		args = append(args, "%"+opt.ListOptions.MatchQuery+"%", "%"+opt.ListOptions.MatchQuery+"%")
+	}
+
+	if opt.ActivityType != "" {
+		activitiesQ += " AND a.activity_type = ?"
+		args = append(args, opt.ActivityType)
+	}
+
+	fmt.Println("opt.StartCreatedAt:", opt.StartCreatedAt)
+	fmt.Println("opt.EndCreatedAt:", opt.EndCreatedAt)
+
+	if opt.StartCreatedAt != "" || opt.EndCreatedAt != "" {
+		start := opt.StartCreatedAt
+		end := opt.EndCreatedAt
+		if start == "" && end != "" {
+			// Only EndCreatedAt is set, so filter up to end
+			activitiesQ += " AND a.created_at <= ?"
+			args = append(args, end)
+		} else if start != "" && end == "" {
+			// Only StartCreatedAt is set, so filter from start to now
+			activitiesQ += " AND a.created_at >= ? AND a.created_at <= ?"
+			args = append(args, start, time.Now())
+		} else if start != "" && end != "" {
+			// Both are set
+			activitiesQ += " AND a.created_at >= ? AND a.created_at <= ?"
+			args = append(args, start, end)
+		}
+	}
+
```

Comment:
Tagging @MagnusHJensen for review the query against applicable index coverage

Reply:
@gillespi314 I think my idea was to either comment this out, remove it or leave it be, and then update all search code to use indexes and add the user search as well in the next story.